lib/progress.ts

```diff
@@ -105,13 +110,13 @@ export function estimate(requestAsync, isBrowser) {
 
 		const stream = require('stream');
```

Comment:
Can you go through and make sure all requires like this have an added `as typeof import...` - in javascript the `require` call itself is typed when possible but in typescript you have to explicitly type it, meaning with the conversion we'll actually have lost typings in some places